common/primitives/src/messages.rs

```diff
@@ -45,6 +45,58 @@ pub struct MessageResponse {
 	#[cfg_attr(feature = "std", serde(skip_serializing_if = "Option::is_none", default))]
 	pub payload_length: Option<u32>,
 }
+
+/// A type for responding with a single Message in an RPC-call dependent on schema model
+/// IPFS, Parquet: { index, block_number, provider_msa_id, cid, payload_length }
+/// Avro, OnChain: { index, block_number, provider_msa_id, msa_id, payload }
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
+pub struct MessageResponseV2 {
```

Comment:
Message storage is now indexed by `IntentId` instead of `SchemaId`. Since multiple Schemas can be stored under one Intent index, we added `schema_id` to the stored payload.